eclair-core/src/main/scala/fr/acinq/eclair/io/ReconnectionTask.scala

```diff
@@ -44,19 +44,19 @@ import scala.util.Random
  * - concurrent incoming/outgoing connections and risk of reconnection loops
  * - etc.
  */
-class ReconnectionTask(nodeParams: NodeParams, remoteNodeId: PublicKey, switchboard: ActorRef, router: ActorRef) extends FSMDiagnosticActorLogging[ReconnectionTask.State, ReconnectionTask.Data] {
+class ReconnectionTask(nodeParams: NodeParams, remoteNodeId: PublicKey) extends FSMDiagnosticActorLogging[ReconnectionTask.State, ReconnectionTask.Data] {
 
   import ReconnectionTask._
 
   startWith(IDLE, IdleData(Nothing))
 
   when(CONNECTING) {
-    case Event(Terminated(actor), d: ConnectingData) if actor == d.connection =>
+    case Event(Status.Failure(_: Client.ConnectionFailed), d: ConnectingData) =>
       log.info(s"connection failed, next reconnection in ${d.nextReconnectionDelay.toSeconds} seconds")
       setReconnectTimer(d.nextReconnectionDelay)
       goto(WAITING) using WaitingData(nextReconnectionDelay(d.nextReconnectionDelay, nodeParams.maxReconnectInterval))
 
-    case Event(FSM.Transition(_, Peer.DISCONNECTED, Peer.CONNECTED), d) =>
+    case Event(Peer.Transition(_, _: Peer.ConnectedData), d) =>
```

Comment:
I don't understand how we can receive this event now...reconnection task doesn't listen to the peer's FSM transitions anymore, and the peer only explicitly sends a transition to the reconnection task when going to `DISCONNECTED`...

Reply:
You're right, we aren't.

Reply:
Fixed with 9c6eed3.